src/Core/Domain/ValueObject/Status.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+
+/**
+ * Holds data of object status
+ */
+class Status
```

Comment:
@matks Extracted this class, could you take a look please?

Reply:
Code is 👍 but I need @eternoendless opinion about namespace. It's important because you introduce the 1st ValueObject in the root Domain folder, not in a dedicated Domain folder.

Reply:
@zuk3975 All right, I got a relevant answer from @eternoendless 😄 

For the namespace: it's all right, when we need global ValueObjects it's OK to put them in `src/Core/Domain/ValueObject`

However he disagrees with the Status class. According to him, the goal of using ValueObjects is to kindof create new php types when they are not available. Php provides types `integer`, `float`, `string` ... but we might need some custom types. Email is one subtype of a string: it's a string that is only valid if its content complies with RFC for emails. It's useful to have it and use it.

However for Status we have only redefined a boolean here. If the Status VO could take some more values such as "open", "closed", "deleted", "unknown", maybe it could be useful. But instead we have a created a redundant type, a copy of `boolean`.

So can we not use Status 😉 please ?